### PR TITLE
[TGUI] Mulebot TGUI

### PR DESCRIPTION
**Pull request by Oyu07**
## What Does This PR Do

Mulebot joins the TGUI club

//TODO

- [ ] Add Report Delivery to status

Still working on it, making draft PR for suggestions and putting it out there

## Why It's Good For The Game

Last bot the be TGUI

## Images of changes

<details>
<summary>OLD</summary>

| Locked                                                                                                       | Unlocked                                                                                                     |
| ------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------ |
| Ivzex1XcGS | gbqoEAV09E |
</details>

<details>
<summary>TGUI</summary>

+ <details>
    <summary>Status</summary>

    | LOCKED                                                                                                       | UNLOCKED                                                                                                     |
    | ------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------ |
    | W3O9OJsj5F | uZf0bhOsGZ |
    </details>

+ <details>
    <summary>Cargo</summary>

    | LOCKED                                                                                                       | UNLOCKED                                                                                                     |
    | ------------------------------------------------------------------------------------------------------------ | ------------------------------------------------------------------------------------------------------------ |
    | VVrf6RCEsw | orGVGGzk9Q |
    </details>

+ <details>
    <summary>No Battery</summary>

    KkvwuXJb9B
    </details>

</details>

## Testing

- [ ] Latest Commit Tested
- [ ] AI

  - Set Destination
  - Go / Home / Stop
  - Wires

- [ ]  Human
  - Set Destination
  - Go / Home / Stop
  - Wires

- [ ] Borgo
  - Set Destination
  - Go / Home / Stop
  - Wires

## Changelog
:cl:
tweak: Mulebot TGUI
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 1]
```diff
@@ -262,15 +281,100 @@
...
-		return 1
+
+/mob/living/simple_animal/bot/mulebot/show_controls(mob/user)
+	if(!open)
+		ui_interact(user)
 	else
-		to_chat(user, "<span class='danger'>Access denied.</span>")
-		return 0
+		if(isAI(user))
+			to_chat(user, "<span class='warning'>The bot is in maintenance mode and cannot be controlled</span>")
```
Comment (Burzah) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm:293]:
```suggestion
			to_chat(user, "<span class='warning'>The bot is in maintenance mode and cannot be controlled.</span>")
```

File: tgui/packages/tgui/interfaces/BotMule.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  const { act, data } = useBackend(context);
+  const [tabIndex, setTabIndex] = useLocalState(context, 'tabIndex', 0);
+  const decideTab = (index) => {
+    switch (index) {
+      case 0:
+        return <BotStatus />;
+      case 1:
+        return <MuleLoad />;
+      default:
+        return 'Whoops!!';
```
Comment (Burzah) [inline tgui/packages/tgui/interfaces/BotMule.js:24]: Can we have this return a string instead that is along the lines of "Whoops! Something went wrong! Contact a coder." or something like that?

**Review by Burzah — COMMENTED** [review]
Just at a quick glance, even though this PR is still a work in progress.

**Comment by Oyu07** [comment]
This is going to be stale / on hold for a bit, Just gotta get the energy to finish what I plan with it

**Comment by Oyu07** [comment]
!merge_upstream

**Pull request closed without merging**